docs(tools): add documentation for the deno lsp command (#8676)
denoland/deno: docs/getting_started/setup_your_environment.md

}
```

#### LSP clients

Deno has builtin support for the
[Language server protocol](https://langserver.org).

> If your editor supports the LSP, you can add Deno as a language server for

Reviewer: ```suggestion
If your editor supports the LSP, you can use Deno as a language server for
```

TypeScript and JavaScript.

> The editor can start the server with `deno lsp`.

Reviewer: ```suggestion
The language server can be started with `deno lsp`.
```

Author: Eh, I feel like people that do not know how the LSP works will interpret this as "Type `deno lsp` into the terminal".

Reviewer: Sure

> 

Reviewer 2: It would be great if we could include example for other popular editor (vim/sublime). CC @uki00a I believe you set up lsp with vim?

Reviewer 3: Yeah, I set up `deno lsp` with [`vim-lsp`](https://github.com/prabirshrestha/vim-lsp) like this:

```markdown
##### Example for Vim/Neovim

After installing the [`vim-lsp`](https://github.com/prabirshrestha/vim-lsp) LSP client
you can add the Deno language server by adding the following to your
`vimrc`/`init.vim`:

```vim
if executable("deno")
  augroup LspTypeScript
    autocmd!
    autocmd User lsp_setup call lsp#register_server({
    \ "name": "deno lsp",
    \ "cmd": {server_info -> ["deno", "lsp"]},
    \ "root_uri": {server_info->lsp#utils#path_to_uri(lsp#utils#find_nearest_parent_file_directory(lsp#utils#get_buffer_path(), "tsconfig.json"))},
    \ "whitelist": ["typescript", "typescript.tsx"],
    \ })
  augroup END
endif
```
```

Reviewer 4: Was just about to write a PR to add this to the docs actually so great if we can include this here (also use vim-lsp).

Altho I wouldn't use tsconfig as the root, because most Deno projects don't have one.

##### Example for Kakoune

After installing the [`kak-lsp`](https://github.com/kak-lsp/kak-lsp) LSP client
you can add the Deno language server by adding the following to your
`kak-lsp.toml`

```toml
[language.deno]

> filetypes = ["typescript", "javascript"]

Reviewer 5: ```suggestion
filetypes = ["typescript", "typescriptreact", "javascript", "javascriptreact"]
```

Author: Kakoune uses `typescript` and `javascript` for `.jsx` and `.tsx` files.
https://github.com/mawww/kakoune/blob/ce96c3f26246ca1d1ae380e5a14e2880256ded25/rc/filetype/javascript.kak#L4-L10
